### PR TITLE
add github action to assign milestone automatically

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
None

### Summary of the issue:
NV Access has a server side script to automatically assign milestones to closed issues and PRs. This script has been malfunctioning for some time meaning we are not tracking which release issues are closed in.

### Description of development approach
Used copilot to generate a github action to assign milestones.

### Testing strategy
Manual testing
- Succesful milestone update:
  - https://github.com/nvaccess/nvda-test/issues/3
  - https://github.com/nvaccess/nvda-test/actions/runs/10211627120/job/28253321745
- Retaining previous milestone value if already set
  - https://github.com/nvaccess/nvda-test/issues/1
  - https://github.com/nvaccess/nvda-test/actions/runs/10211649094/job/28253382737

### Known issues with pull request:
- [ ] Need to update internal release docs to ensure the milestone id is updated when starting the release cycle:  https://github.com/nvaccess/Internal-issue-tracking/pull/538
- [x] need to tear out dead automilestone script from server - https://github.com/nvaccess/nvaccessServer/issues/57

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
	- Introduced an automated workflow that assigns a default milestone to closed issues and pull requests for improved project management and organization.

### CODE REVIEW

File: .github/workflows/assign-milestone-on-close.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Assign Milestone on Close
+
+env:
+  MILESTONE_ID: 1
```
Comment (seanbudd) [inline .github/workflows/assign-milestone-on-close.yml:4]:
```suggestion
  MILESTONE_ID: 80
```

File: .github/workflows/assign-milestone-on-close.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Assign Milestone on Close
+
+env:
+  MILESTONE_ID: 80  # 2024.4
```
Comment (seanbudd) [inline .github/workflows/assign-milestone-on-close.yml:4]:
```suggestion
  MILESTONE_ID: ${{ secrets.MILESTONE_ID }}
```

File: .github/workflows/assign-milestone-on-close.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: Assign Milestone on Close
+
+env:
+  MILESTONE_ID: 1
+
```
[the remark this answers is not in the document]
Comment (seanbudd) [reply .github/workflows/assign-milestone-on-close.yml]: this should be an environment variable not a secret

[the remark this answers is not in the document]
Comment (seanbudd) [reply .github/workflows/assign-milestone-on-close.yml]: can you make this so it's an environment variable set through GitHub, not hard coded

File: .github/workflows/assign-milestone-on-close.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+name: Assign Milestone on Close
+
+env:
+  MILESTONE_ID: ${{ secrets.MILESTONE_ID }}
```
Comment (seanbudd) [inline .github/workflows/assign-milestone-on-close.yml:4]:
```suggestion
  MILESTONE_ID: ${{ vars.MILESTONE_ID }}
```

**Review by SaschaCowley — COMMENTED** [review]
This looks reasonable to me, though I've never worked on GitHub scripting before, so I can't be confident.

**Comment by XLTechie** [comment]
@seanbudd
Instead of depending on a human to set the milestone, can the milestone be derived from either the repo or git log?

Since an Action can set repo variables, I'm thinking that a separate workflow to run on every merge, to update the milestone if it should be changed, might be possible.

https://github.com/mmoyaferrer/set-github-variable

**Comment by seanbudd** [comment]
@XLTechie - I don't think so, no. We want to update this manually when we update `build_version` and start the new release cycle. I don't envision a simple programmatic way to determine if the PR we are merging is starting a new release cycle.

**Review by gerald-hartig — APPROVED** [review]

**Pull request merged**